### PR TITLE
[ONPREM-1052] Add PVC and mount to agent pod

### PR DESCRIPTION
**Pull request by Stuart133**
:gear: **Issue**

*Ticket/s*:   

:gear: **Change** 

Change Type: 

AC:

:white_check_mark: **Fix**

:question: **Tests**

- [ ] Tests for new feature and update for changes
- [ ] Linting passes and/or formatting matches the standard of the repo

🗒️ **Documentation**

- [ ] Updated related documentation (if applicable).
- [ ] Updated Change log (if exists)

### CODE REVIEW

File: templates/deployment.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ spec:
     {{- if .Values.agent.pullSecrets }}
       imagePullSecrets:
       {{- range .Values.agent.pullSecrets }}
-        - {{ toYaml . }}
+        - name: {{ toYaml . }}
```
Comment (Stuart133) [inline templates/deployment.yaml:27]: This was broken before

File: values.yaml [hunk 2]
```diff
@@ -25,7 +25,8 @@ agent:
   podAnnotations: {}
 
   # -- Security Context policies for agent pods
-  podSecurityContext: {}
+  podSecurityContext:
```
Comment (Stuart133) [inline values.yaml:28]: I will move this to a template when cleaning this PR up. It's needed to allow the `circleci` user in the agent container write permissions to the mounted volume

File: values.yaml [hunk 3]
```diff
@@ -262,6 +261,25 @@ agent:
     # Note that the number of concurrent jobs rerun using SSH will be limited by the size of this range.
     numPorts: 20
 
+  # -- Use a volume to store task related binaries to avoid copying into the task pod filesystem for
```
Comment (christian-stephen) [inline values.yaml:264]: Can we add a unit test for this functionality?

**Review by christian-stephen — APPROVED** [review]
LGTM, provided we add some unit tests

**Pull request merged**